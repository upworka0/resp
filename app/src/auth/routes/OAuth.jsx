import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Progress } from 'reactstrap';
import { Redirect } from 'react-router-dom';

export default class OAuth extends Component {
  state = {
    hasValidationBeenFailed: false,
    redirect: false,
  };

  componentDidMount() {
    const { validateToken, location } = this.props;

    const params = new URLSearchParams(location.search);
    const accessToken = params.get('access');
    const refreshToken = params.get('refresh');

    validateToken(accessToken, refreshToken)
      .then(() => {
        this.setState({ redirect: true });
      })
      .catch(() => {
        this.setState({ hasValidationBeenFailed: true });
      });
  }

  render() {
    const { hasValidationBeenFailed, redirect } = this.state;

    if (hasValidationBeenFailed) {
      return (<Redirect to="/signin" />);
    }

    if (redirect) {
      return (<Redirect to="/" />);
    }

    return (
      <section className="d-flex flex-column flex-grow-1 align-items-center justify-content-center">
        <div className="px-4 py-3 shadow-z3 rounded">
          <h5 className="text-center px-5">Verifying your token...</h5>
          <Progress animated value={100} />
        </div>
      </section>
    );
  }
}

OAuth.propTypes = {
  validateToken: PropTypes.func.isRequired,
  location: PropTypes.object.isRequired,
};
