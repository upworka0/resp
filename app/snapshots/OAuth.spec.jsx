import React from 'react';
import { shallow } from 'enzyme';

import OAuth from '../src/auth/routes/OAuth';

describe('OAuth UI component', () => {
  let wrapper;
  const validateTokenMock = jest.fn();
  const location = {};

  beforeEach(() => {
    validateTokenMock.mockResolvedValueOnce();
    wrapper = shallow(<OAuth validateToken={validateTokenMock} location={location} />);
  });

  it('should render progress bar', () => {
    wrapper.setState({
      hasValidationBeenFailed: false,
      redirect: false,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render redirect to sign in if authentication failed', () => {
    wrapper.setState({ hasValidationBeenFailed: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render redirect to root if authentication succeed', () => {
    wrapper.setState({ redirect: true });
    expect(wrapper).toMatchSnapshot();
  });
});
