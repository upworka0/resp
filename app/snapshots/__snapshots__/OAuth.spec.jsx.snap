// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OAuth UI component should render progress bar 1`] = `
<section
  className="d-flex flex-column flex-grow-1 align-items-center justify-content-center"
>
  <div
    className="px-4 py-3 shadow-z3 rounded"
  >
    <h5
      className="text-center px-5"
    >
      Verifying your token...
    </h5>
    <Progress
      animated={true}
      max={100}
      tag="div"
      value={100}
    />
  </div>
</section>
`;

exports[`OAuth UI component should render redirect to root if authentication succeed 1`] = `
<Redirect
  push={false}
  to="/"
/>
`;

exports[`OAuth UI component should render redirect to sign in if authentication failed 1`] = `
<Redirect
  push={false}
  to="/signin"
/>
`;
